source/PopNGo/PopNGo/PopNGo/wwwroot/js/favorites.js

```diff
@@ -0,0 +1,64 @@
+import { formatStartTime } from './Helper-Functions/formatStartTime.js';
+
+
+function fetchUserFavorites() {
+    fetch('/api/FavoritesApi/GetUserFavorites')
+        .then(response => {
+            if (response.status === 401) {
+                // Unauthorized, tell the user to log in or sign up
+                document.getElementById('login-prompt').style.display = 'block';
+                throw new Error('Unauthorized');
+            }
+            if (!response.ok) {
```

Comment:
Testing this myself, this should have an `else if` before `if (response.ok)` to check for response status 404 to be able to show the no-favorites-message and either throw an error or return.

Without adding it here, the code sees that the response is not ok and always aborts, so there is no notification about not having any favorites.